### PR TITLE
Update version compat line in README

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

**Review by arvindkrishnakumar-okta — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Okta Spring Boot Starter
 
 Okta's Spring Boot Starter will enable your Spring Boot application to work with Okta via OAuth 2.0/OIDC.
 
-**NOTE:** This library works with Spring Boot 2.2+. If you need support for Spring Boot 1.5.x, use version 0.6.
+**NOTE:** This library works with Spring Boot 2.4+. If you need support for Spring Boot 1.5.x, use version 0.6.
```
Comment (arvindkrishnakumar-okta) [inline README.md:11]: @bdemers should we add another line that says `if you need support for Spring Boot 2.2.x, use version 1.4.0`?

Comment (bdemers) [reply README.md:11]: Yeah, maybe we should break it out into a grid, maybe something similar to our other READMEs "release" section:
https://github.com/okta/okta-sdk-java#release-status

Comment (arvindkrishnakumar-okta) [reply README.md:11]: updated.

File: README.md [hunk 2]
```diff
@@ -8,7 +8,32 @@ Okta Spring Boot Starter
...
+
+## Release status
+
+This library uses semantic versioning and follows Okta's [library version policy](https://developer.okta.com/code/library-versions/).
+
+:heavy_check_mark: The current stable major version series is: 1.5.x
```
Comment (bdemers) [inline README.md:17]: We can probably just say `1.x`, it's only the major versions that fall under the "Okta library version policy"

File: README.md [hunk 3]
```diff
@@ -8,7 +8,32 @@ Okta Spring Boot Starter
...
+
+| Version | Status                    |
+| ------- | ------------------------- |
+| 0.x.x | :warning: Retired |
+| 1.0.0 - 1.3.0 | :warning: Retired |
+| 1.4.0 | :clock9: Retiring effective August 20, 2021 |
+| 1.5.0 | :heavy_check_mark: Stable |
```
Comment (bdemers) [inline README.md:24]:
```suggestion
| 0.x.x | :warning: Retired |
| 1.x.0 | :heavy_check_mark: Stable |
```

See above comment

File: README.md [hunk 4]
```diff
@@ -8,7 +8,32 @@ Okta Spring Boot Starter
...
+
+## Spring Boot Compatibility
+
+| Okta Spring Boot SDK Version | Compatible Spring Boot Version |
+| ------- | ------------------------- |
+| 0.6 | 1.5.x |
+| 1.0.0, 1.1.0 | 2.1 |
+| 1.2.x | 2.2.0.M3 |
+| 1.3.0, 1.4.0 | 2.2.0.RELEASE |
+| 1.5.0 | 2.4.0 |
```
Comment (bdemers) [inline README.md:34]: We could just copy the ranges from: https://github.com/spring-io/start.spring.io/blob/17470056557679f0641f6bf653bcddacc53eb962/start-site/src/main/resources/application.yml#L427-L439

File: README.md [hunk 5]
```diff
@@ -25,12 +25,10 @@ This library uses semantic versioning and follows Okta's [library version policy
...
-| 0.6 | 1.5.x |
-| 1.0.0, 1.1.0 | 2.1 |
-| 1.2.x | 2.2.0.M3 |
-| 1.3.0, 1.4.0 | 2.2.0.RELEASE |
-| 1.5.0 | 2.4.0 |
- 
+| 1.2.1 | 2.1.2.RELEASE, 2.2.0.M1 |
+| 1.4.0 | 2.2.0.M1, 2.4.0-M1 |
+| 1.5.0 | 2.4.0-M1, 2.5.0-M1 |
+
```
Comment (bdemers) [inline README.md:29]: We can leave out the `M` releases as those are milestones and are not supported versions, instead, we could just use ranges, either maven range syntax, `>=`, or maybe just something like `2.2.x - 2.3.x`?

Thoughts?

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Pull request merged**